relay-server/src/actors/global_config.rs

```diff
@@ -256,13 +261,16 @@ impl Service for GlobalConfigService {
                 relay_log::info!("fetching global configs disabled: no credentials configured");
             }
 
+            let mut shutdown = false;
```

Comment:
I don't really like this variable and I'd rather change the internal state (e.g. resetting the fetch handle) and let branches not resolve. However, this approach would still continue processing internal messages*, which I've tried to avoid with this solution.

*I've not found any way to drop all the remaining messages in the channel in constant time, and blocking on that seems a waste of service availability (closing the channel doesn't drop the messages).